tests/functional/u/unnecessary/unnecessary_dunder_call.py

```diff
@@ -27,12 +42,39 @@ class Bar2(object):
     def __new__(cls):
         super().__new__(cls)
 
-class Base:
-    @classmethod
-    def get_first_subclass(cls):
-        for subklass in cls.__subclasses__():
-            return subklass
-        return object
+class CustomRegistry(dict):
+    def __init__(self) -> None:
+        super().__init__()
+        self._entry_ids = {}
+
+    def __setitem__(self, key, entry) -> None:
+        super().__setitem__(key, entry)
+        self._entry_ids.__setitem__(entry.id, entry)
```

Comment:
Could you add a test case where the call function name is different to the parent function name? I.e. `__delitem__` inside `__setitem__`.